### PR TITLE
6290 dataset role mgmt api

### LINKED ISSUE #6290: Dataset Role Management API
As Dataverse user, I would like to assign a specific role/user to a specific dataset using the Native Api. As we can do with Dataverse. I didn't found it in the documentation.

Thanks in advance;
Flavio.

### PR DESCRIPTION
**Pull request by sekmiller**
**What this PR does / why we need it**:
Allows admins to assign roles to users at the dataset level

**Which issue(s) this PR closes**:

Closes #6290 Dataset Role Management API

**Special notes for your reviewer**:
None
**Suggestions on how to test this**:

**Does this PR introduce a user interface change?**:
None
**Is there a release notes update needed for this change?**:
No

**Additional documentation**:

### CODE REVIEW

File: src/main/java/edu/harvard/iq/dataverse/api/Datasets.java [hunk 1]
```diff
@@ -1084,24 +1086,38 @@ public Response getLinks(@PathParam("id") String idSupplied ) {
...
+                    if (aRole.getAlias().equals(ra.getRole())) {
+                        theRole = aRole;
+                        break;
+                    }
+                }
+                dv = dv.getOwner();
             }
-            DataverseRole theRole = rolesSvc.findBuiltinRoleByAlias("admin");
+            if (theRole == null) {
+                return error(Status.BAD_REQUEST, "Can't find role named '" + ra.getRole() + "' in dataverse " + dataset.getOwner());
```
Comment (pdurbin) [inline src/main/java/edu/harvard/iq/dataverse/api/Datasets.java]: I think the new-ish rule is that when we touch APIs we should put the English in a bundle.

Comment (sekmiller) [reply src/main/java/edu/harvard/iq/dataverse/api/Datasets.java]: Added revoke and api messages to the Bundle.

**Comment by kcondon** [comment]
@sekmiller Need GET after -X in list assignments example.

**Comment by kcondon** [comment]
@sekmiller Discussed with @scolapasta and @djbrooke , please add inherited as well as directly assigned roles and indicate which is which.

**Comment by djbrooke** [comment]
Thanks @kcondon for the discussion and @sekmiller for taking another pass at this!

**Pull request merged**